src/main/java/com/greenfoxacademy/goddesstribesbackend/controllers/BuildingController.java

```diff
@@ -0,0 +1,114 @@
+package com.greenfoxacademy.goddesstribesbackend.controllers;
+
+import com.greenfoxacademy.goddesstribesbackend.models.MockData;
+import com.greenfoxacademy.goddesstribesbackend.models.dtos.*;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class BuildingController {
+
+  @ApiImplicitParams({
+      @ApiImplicitParam(name = "token", value = "Authorization token",
+          required = true, dataType = "string", paramType = "header") })
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message ="OK", response = BuildingDTO.class)})
+  @GetMapping(value = "/kingdom/buildings",
+  produces = { "application/json" },
+  consumes = { "application/json" })
+  public ResponseEntity<Object> mockListOfBuildings() {
+    return ResponseEntity.status(200).body(MockData.buildingsDTO);
+  }
+
+  @ApiImplicitParams({
+      @ApiImplicitParam(name = "token", value = "Authorization token",
+          required = true, dataType = "string", paramType = "header") })
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message ="OK", response = BuildingDTO.class),
+      @ApiResponse(code = 400, message ="Missing parameter(s): type!", response = ErrorMessage.class),
+      @ApiResponse(code = 406, message ="Invalid building type", response = ErrorMessage.class),
+      @ApiResponse(code = 409, message ="Not enough resource", response = ErrorMessage.class)})
```

Comment:
We could also use status codes from constants. I bet that ApiResponse class have them, did not intellisense suggest them?

Reply:
Refactrored to one line

Reply:
refactored...